src/test/java/org/fenixedu/academic/domain/degreeStructure/ProgramConclusionConfigTest.java

```diff
@@ -0,0 +1,184 @@
+            DegreeTest.initDegree();
+            degreeCurricularPlan = new DegreeCurricularPlan(Degree.find(DEGREE_A_CODE), DCP_NAME_V1, AcademicPeriod.THREE_YEAR);
+            programConclusion = new ProgramConclusion();
+
+            return null;
+        });
+    }
+
+    @After
+    public void cleanup() {
+        Bennu.getInstance().getProgramConclusionConfigsSet().forEach(ProgramConclusionConfig::delete);
+    }
+
+    private ProgramConclusionConfig create(LocalizedString conclusionTitle, DegreeCurricularPlan degreeCurricularPlan,
+            ProgramConclusion programConclusion) {
+        return ProgramConclusionConfig.create(conclusionTitle, degreeCurricularPlan, programConclusion);
+    }
+
+    @Test
+    public void testProgramConclusionConfig_createSuccess() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+
+        assertNotEquals(null, config);
+        assertEquals(CONCLUSION_TITLE, config.getConclusionTitle());
+        assertEquals(degreeCurricularPlan, config.getDegreeCurricularPlan());
+        assertEquals(programConclusion, config.getProgramConclusion());
+    }
+
+    @Test
+    public void testProgramConclusionConfig_setNullConclusionTitleThrowsDomainException() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+        assertThrows(DomainException.class, () -> config.setConclusionTitle(null));
+    }
+
+    @Test
+    public void testProgramConclusionConfig_setEmptyConclusionTitleThrowsDomainException() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+        assertThrows(DomainException.class, () -> config.setConclusionTitle(new LocalizedString(Locale.getDefault(), "")));
+    }
+
+    @Test
+    public void testProgramConclusionConfig_setNullDegreeCurricularPlanThrowsDomainException() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+        assertThrows(DomainException.class, () -> config.setDegreeCurricularPlan(null));
+    }
+
+    @Test
+    public void testProgramConclusionConfig_setNullProgramConclusionThrowsDomainException() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+        assertThrows(DomainException.class, () -> config.setProgramConclusion(null));
+    }
+
+    @Test
+    public void testProgramConclusionConfig_setConfigOrderThrowsDomainException() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+        assertThrows(DomainException.class, () -> config.setConfigOrder(1));
+    }
+
+    @Test
+    public void testProgramConclusionConfig_changeConfigOrder() {
+        config = create(CONCLUSION_TITLE, degreeCurricularPlan, programConclusion);
+        int initialOrder = config.getConfigOrder();
+        config.changeConfigOrder(2);
+        assertEquals(2, config.getConfigOrder());
+        assertNotEquals(initialOrder, config.getConfigOrder());
+    }
```

Comment:
Is it possible to add tests for all the move scenarios? (moveUp, moveTop, etc..)